src/lib/homestore_backend/hs_blob_manager.cpp

```diff
@@ -12,17 +12,21 @@ static constexpr uint64_t io_align{512};
 BlobManager::AsyncResult< blob_id_t > HSHomeObject::_put_blob(ShardInfo const& shard, Blob&& blob) {
     auto& pg_id = shard.placement_group;
     shared< homestore::ReplDev > repl_dev;
+    blob_id_t new_blob_id;
     {
         std::shared_lock lock_guard(_pg_lock);
         auto iter = _pg_map.find(pg_id);
         RELEASE_ASSERT(iter != _pg_map.end(), "PG not found");
         repl_dev = static_cast< HS_PG* >(iter->second.get())->repl_dev_;
+        new_blob_id = iter->second->blob_sequence_num_.fetch_add(1, std::memory_order_relaxed);
+        RELEASE_ASSERT(new_blob_id < std::numeric_limits< decltype(new_blob_id) >::max(),
```

Comment:
How can `new_blob_id` hold a value larger than it itself can hold? Was this meant to be `decltype(blob_sequence_num_) > ::max()`?